### DISCUSSION TITLE
A question about annotation

### DISCUSSION THREAD

**Discussion by xiaowenchenjax** [stuart-lab/signac#669] [Q&A] [answered]
Hi authors,
my scATAC-seq data were aligned to the reference genome "gencode vM17", I looked through the previous discussions about it, I seems like I need annotate scATAC-seq data by this annotation file. I can provide TSS position when running TSSEnrichment() function, when calculating gene activity, i need get the coordinates of genes using the following codes:
gtf <- rtracklayer::import('genes.gtf')
gene.coords <- gtf[gtf$type == 'gene']
seqlevelsStyle(gene.coords) <- 'UCSC'
gene.coords <- keepStandardChromosomes(gene.coords, pruning.mode = 'coarse')

My question is why we need sequence level is UCSC? Do I need to run this script when i used the reference from gencode? if not, what sequence level I need to assign? 

Thank you.

**Comment by xiaowenchenjax** [comment]
the sequence level is UCSC that means chromosomes like this chr1, chr2...

**Comment by timoast** [comment, marked as answer]
You need to set the `seqlevelsStyle` for you gene annotations to match whatever genome the data were mapped to. This will not always be UCSC, that is just given as an example on the Signac website.

**Reply by xiaowenchenjax** [reply]
Thanks, based on the following the script 
annotations <- GetGRangesFromEnsDb(ensdb = EnsDb.Mmusculus.v79)
mm10.tss<-GetTSSPositions(annotations)
Can I get the correct TSS position? seems like annotations contain the information of all the exon,gap , utr , cds  for one gene. 
What is gap? Based on discussion  "GeneActivity function and TSS location #508", you collapse each gene down to the longest transcript and take the start of that transcript as the TSS for the gene. 
when I check the script, for one gene, i found TSS is the min start value of  exon, gap,  utr,  cds if strand = +, otherwise, TSS is max end of all exon, gap,  utr,  cds. but transcripts is not composed of exon, gap, utr, cds randomly. 
Is this method of calculating TSS correct? or I miss something?
